src/test/java/no/hvl/dat250/h2020/group5/service/VoteServiceTest.java

```diff
@@ -0,0 +1,132 @@
+package no.hvl.dat250.h2020.group5.service;
+
+import no.hvl.dat250.h2020.group5.dao.PollRepository;
+import no.hvl.dat250.h2020.group5.dao.VoteRepository;
+import no.hvl.dat250.h2020.group5.dao.VoterRepository;
+import no.hvl.dat250.h2020.group5.entities.*;
+import no.hvl.dat250.h2020.group5.requests.CastVoteRequest;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
```

Comment:
Bra bruk av mocking! 🥇 Ifølge [https://stackoverflow.com/questions/55276555/when-to-use-runwith-and-when-extendwith](url) så kan man bruke @SpringBootTest istedenfor, kanskje vi burde bruke det for at det skal være lite variasjon i koden. :)

Reply:
Sant at det fungerer med begge deler :) Grunnen til at jeg brukte ExtendWith() var fordi det var mye kjappere. I stedet for å laste hele applikasjonen med @SpringBootTest, trenger den bare deler av den.

Reply:
Okay! Kanskje vi burde bruke @ExtendWith flere steder da? Hvordan vet man at man bare trenger deler av den? 😮

Reply:
Siden jeg bare mocker repositoriene så trenger vi ikke at applikasjonen starter fullt og helt. Mens i adminkontrolleren har jeg testet at hele systemet fungerer som det skal, fra kontroller til repository, og da trenger man SpringBootTest

Reply:
Aha, smart! 👍